client/src/views/logs.tsx

```diff
@@ -77,9 +81,38 @@ export default class Logs extends Base<Props, State> {
         });
     }
 
+    startDownloadLog = () => {
+        const {namespace, name} = this.props;
+        const {container, showPrevious = false, logDownloading} = this.state;
+
+        if (!container || logDownloading) {
+            return;
+        }
+
+        this.setState({logDownloading: true}, () => {
+            api.logsAll(namespace, name, container, showPrevious).then((message) => {
```

Comment:
Any reason to not use the `async/await` syntax here? I think it would be more readable and we've generally tried to avoid the `.then` syntax throughout the codebase.

Reply:
Replaced with async/await